ultralytics/engine/exporter.py

```diff
@@ -273,6 +290,18 @@ def __call__(self, model=None) -> str:
             elif isinstance(m, C2f) and not is_tf_format:
                 # EdgeTPU does not support FlexSplitV while split provides cleaner ONNX graph
                 m.forward = m.forward_split
+            if isinstance(m, Detect) and imx:
+                from ultralytics.utils.tal import make_anchors
+
+                m.anchors, m.strides = (
+                    x.transpose(0, 1)
+                    for x in make_anchors(
+                        torch.cat([s / m.stride.unsqueeze(-1) for s in self.imgsz], dim=1), m.stride, 0.5
+                    )
+                )
```

Comment:
@ambitious-octopus @Laughing-q why are anchors being manually computed here when they are computed in the forward pass anyways?

Reply:
@glenn-jocher Sony export is based on torch.fx. They use torch.fx to export the static graph of the model and do quantization based on the graph. And unfortunately we need this to make torch.fx work.
Rules for torch.fx to export the static graph is that: 
- It takes the input as `symbol` rather than a real data, so we're not able to access any of the attributes from input. And our anchors generation in head.py is using its dtype and device so it does not work with it.
  - To resolve and I tried not using dtype and device in the function but for that we need to explicitly put the generated anchors to correct device which we can't get the correct device in this Detect.forward method because we can not access attributes like `device` of the input.
- We also can not use dynamic control flow(where the condition may depend on the value of input, again it's a symbol, not real number so can not have its true value). See https://pytorch.org/docs/stable/fx.html#dynamic-control-flow.
  - which mean https://github.com/ultralytics/ultralytics/pull/14878/commits/50aba1e908dbcd2e5d77ccd45f45e70b71a7c97a has already broken the workflow, since we need the format at the front so it doesn't access `x[0].shape` when export imx format.

Reply:
@glenn-jocher I've reverted the updates that breaks the workflow in https://github.com/ultralytics/ultralytics/pull/14878/commits/4c36795116779ef8407f6a6188d9111bbc58dcb2 for now